homeassistant/components/template/cover.py

```diff
@@ -254,7 +255,7 @@ async def async_added_to_hass(self):
     @callback
     def _update_state(self, result):
```

Comment:
Since we only care about string values, why would we ever get non-string values?

Reply:
After some digging, realized that this is a task out of scope for this PR.

Reply:
Well, we will never know how the template is crafted, so a non-string value can be a result.

I did notice this changed behavior, as it should have resulted in a `0` value position, not a `NoneType` value. So adjusted that.